include/dt-bindings/pinctrl/stm32-pinctrl.h

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _STM32_PINCTRL_H_
+#define _STM32_PINCTRL_H_
+
+#ifdef CONFIG_SOC_SERIES_STM32F1X
```

Comment:
rather than using CONFIG_* lets split this into different files:
stm32-f1x-pinctrl.h
stm32-f3-f4-l4-pinctrl.h

As mentioned before that the headers in here should be standalone from Zephyr